### PR TITLE
35856 Display Canceled requests - Pending View

### PR DESCRIPTION
**Pull request by vbahinwillit**
## Description
This PR displays pending and canceled appointments grouped

## Original issue(s)
department-of-veterans-affairs/va.gov-team#35856

## Testing done
Unit testing

## Screenshots
localhost_3001_health-care_schedule-view-va-appointments_appointments_requested

## Acceptance criteria
- [ ] All test should pass

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/appointment-list/components/RequestedAppointmentsListGroup.jsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+      temp[groupByKey].push(currentValue);
+      return temp;
+    },
+    {},
+  );
+
+  // Next, get an array of [key, value] pairs. Sort the array in decending order
+  // so that pending appointments are displayed 1st
+  appointmentsByStatus = Object.entries(appointmentsByStatus).sort((a, b) => {
+    if (a[0].toLowerCase() < b[0].toLocaleLowerCase()) return 1;
```
Comment (ptouchton) [inline src/applications/vaos/appointment-list/components/RequestedAppointmentsListGroup.jsx:97]: @vbahinwillit - I am curious about toLowerCase and toLocaleLowerCase usage...what is the need to use the compares that way?

Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/RequestedAppointmentsListGroup.jsx:97]: Good catch. I only intended to use 'toLowerCase'.

File: src/applications/vaos/appointment-list/components/RequestedAppointmentsListGroup.jsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    if (a[0].toLowerCase() > b[0].toLocaleLowerCase()) return -1;
+    return 0;
+  });
+
+  return (
+    <>
+      <div aria-live="assertive" className="sr-only">
```
Comment (timwright12) [inline src/applications/vaos/appointment-list/components/RequestedAppointmentsListGroup.jsx:104]:
```suggestion
      <div aria-live="polite" className="sr-only">
```

Comment (timwright12) [reply src/applications/vaos/appointment-list/components/RequestedAppointmentsListGroup.jsx:104]: as a general rule we don't use assertive unless it's an emergency kind of update

File: src/applications/vaos/appointment-list/components/RequestedAppointmentsListGroup.jsx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+                  <>
+                    <h2>Canceled requests</h2>
+                    <p>Your appointment requests that where canceled</p>
+                  </>
+                )}
+                {/* eslint-disable-next-line jsx-a11y/no-redundant-roles */}
+                <ul
+                  className="vads-u-padding-left--0"
+                  data-cy="requested-appointment-list"
+                  role="list"
```
Comment (timwright12) [inline src/applications/vaos/appointment-list/components/RequestedAppointmentsListGroup.jsx:128]: Curious about why we're adding a list role to a list?

**Review by jenniemc — COMMENTED** [review]
May be good to add a unit test when there is no request (neither proposed or canceled) at all.

File: src/applications/enrollment-verification/components/EnrollmentVerificationBreadcrumbs.jsx [hunk 4]
```diff
@@ -6,7 +6,7 @@ export default function EnrollmentVerificationBreadcrumbs() {
     <a href="/" key="home">
       Home
     </a>,
-    <a href="#" key="education-and-training">
+    <a href="/" key="education-and-training">
```
Comment (vbahinwillit) [inline src/applications/enrollment-verification/components/EnrollmentVerificationBreadcrumbs.jsx:9]: Modified due to eslint error preventing checking.

The href attribute requires a valid value to be accessible. Provide a valid, navigable address as the href value. If you cannot provide a valid href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/anchor-is-valid.md  jsx-a11y/anchor-is-valid

File: src/applications/enrollment-verification/components/EnrollmentVerificationBreadcrumbs.jsx [hunk 5]
```diff
@@ -19,20 +19,20 @@ export default function EnrollmentVerificationBreadcrumbs() {
 
   if (page === 'enrollment-history' || page === 'verify-enrollments') {
     breadcrumbs.push(
-      <a href="#" key="enrollment-history">
+      <a href="/" key="enrollment-history">
```
Comment (vbahinwillit) [inline src/applications/enrollment-verification/components/EnrollmentVerificationBreadcrumbs.jsx:22]: Modified due to eslint error preventing checking.

The href attribute requires a valid value to be accessible. Provide a valid, navigable address as the href value. If you cannot provide a valid href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/anchor-is-valid.md  jsx-a11y/anchor-is-valid

File: src/applications/enrollment-verification/components/EnrollmentVerificationBreadcrumbs.jsx [hunk 6]
```diff
@@ -19,20 +19,20 @@ export default function EnrollmentVerificationBreadcrumbs() {
...
         Verify your school enrollments
       </a>,
     );
   } else if (page === 'review-enrollments') {
     breadcrumbs.push(
-      <a href="#" key="review-enrollments">
+      <a href="/" key="review-enrollments">
```
Comment (vbahinwillit) [inline src/applications/enrollment-verification/components/EnrollmentVerificationBreadcrumbs.jsx:28]: Modified due to eslint error preventing checking.

The href attribute requires a valid value to be accessible. Provide a valid, navigable address as the href value. If you cannot provide a valid href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/anchor-is-valid.md  jsx-a11y/anchor-is-valid

File: src/applications/enrollment-verification/components/EnrollmentVerificationBreadcrumbs.jsx [hunk 7]
```diff
@@ -19,20 +19,20 @@ export default function EnrollmentVerificationBreadcrumbs() {
...
         Enrollment verifications
       </a>,
     );
   }
   if (page === 'verify-enrollments') {
     breadcrumbs.push(
-      <a href="#" key="verify-enrollments">
+      <a href="/" key="verify-enrollments">
```
Comment (vbahinwillit) [inline src/applications/enrollment-verification/components/EnrollmentVerificationBreadcrumbs.jsx:35]: Modified due to eslint error preventing checking.

The href attribute requires a valid value to be accessible. Provide a valid, navigable address as the href value. If you cannot provide a valid href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/anchor-is-valid.md  jsx-a11y/anchor-is-valid

File: src/applications/vaos/appointment-list/components/RequestedAppointmentsListGroup.jsx (hunk 2 above)
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/RequestedAppointmentsListGroup.jsx:104]: Fixed. Thanks for the review!

**Comment by jenniemc** [comment]
There should be a message/instructions on how to schedule an online appointment when user has no proposed request but has several canceled request.  Here's the  [Figma design page](https://www.figma.com/file/gCjrlgjlJUwVUb5SlE8i6J/34487-surface-requested-appointments?node-id=46%3A1472) (scroll to the bottom) .

**Comment by cesaresantiago** [comment]
@vbahinwillit : why is this checkin touching our team's code under src/applications/enrollment-verification/ ?

File: src/applications/enrollment-verification/components/EnrollmentVerificationBreadcrumbs.jsx (hunk 4 above)
Comment (cesaresantiago) [reply src/applications/enrollment-verification/components/EnrollmentVerificationBreadcrumbs.jsx:9]: Understood.

**Review by cesaresantiago — APPROVED** [review]

**Review by timwright12 — APPROVED** [review]

**Review by jenniemc — CHANGES_REQUESTED** [review]
Just need to match design spec for no proposed but existing cancel request.

**Review by jenniemc — APPROVED** [review]
LGTM. Fire away!

**Review by timwright12 — APPROVED** [review]

**Review by ptouchton — APPROVED** [review]

**Review by devonjhills — APPROVED** [review]

**Pull request merged**